plotly/tools.py

```diff
@@ -3442,3 +3694,226 @@ def get_dendrogram_traces(self, X, colorscale):
+                        font=dict(color=min_text_color if val <
+                                  (z_max+z_min) / 2
+                                  else max_text_color),
+                        showarrow=False))
+        return annotations
+
+
+class _Table(FigureFactory):
+    """
+    Refer to TraceFactory.create_table() for docstring
+    """
+    def __init__(self, table_text, colorscale, fontcolor, index,
+                 index_title='', **kwargs):
+        from plotly.graph_objs import graph_objs
+        if _pandas_imported and isinstance(table_text, pd.DataFrame):
+            headers = table_text.columns.tolist()
+            table_text_index = table_text.index.tolist()
+            table_text = table_text.values.tolist()
+            table_text.insert(0, headers)
+            if index:
+                table_text_index.insert(0, index_title)
+                for i in range(len(table_text)):
+                    table_text[i].insert(0, table_text_index[i])
+        self.table_text = table_text
+        self.colorscale = colorscale
+        self.fontcolor = fontcolor
+        self.index = index
+        self.x = range(len(table_text[0]))
+        self.y = range(len(table_text))
+
+    def get_table_matrix(self):
+        '''
+        Create z matrix to make heatmap with striped table coloring
+
+        :rtype (list[list]) table_matrix: z matrix to make heatmap with striped
+            table coloring.
+        '''
+        header = [0] * len(self.table_text[0])
+        odd_row = [.5] * len(self.table_text[0])
+        even_row = [1] * len(self.table_text[0])
+        table_matrix = [None] * len(self.table_text)
+        table_matrix[0] = header
+        for i in range(1, len(self.table_text), 2):
+            table_matrix[i] = odd_row
+        for i in range(2, len(self.table_text), 2):
+            table_matrix[i] = even_row
+        if self.index:
+            for array in table_matrix:
+                array[0] = 0
+        return table_matrix
+
+    def get_table_fontcolor(self):
+        '''
+        Fill fontcolor array.
+
+        Table text color can vary by row so this extends a single color or
+        creates an array to set a header color and two alternating colors to
+        create the striped table pattern.
+
+        :rtype (list) all_font_color: list of fontcolors for each row in table.
+        '''
+        if len(self.fontcolor) == 1:
+            all_font_color = self.fontcolor*len(self.table_text)
+        elif len(self.fontcolor) == 3:
+            all_font_color = range(len(self.table_text))
+            all_font_color[0] = self.fontcolor[0]
+            for i in range(1, len(self.table_text), 2):
+                all_font_color[i] = self.fontcolor[1]
+            for i in range(2, len(self.table_text), 2):
+                all_font_color[i] = self.fontcolor[2]
+        elif len(self.fontcolor) == len(self.table_text):
+            all_font_color = self.fontcolor
+        return all_font_color
```

Comment:
:cow2: There's no else here, so it's hard to tell that `all_font_color` *will* be defined. Can this include an `else` statement?